source/adapters/level_zero/adapter.cpp

```diff
@@ -171,8 +209,40 @@ ur_adapter_handle_t_::ur_adapter_handle_t_()
 
       return;
     }
+    // Dynamically load the new L0 SysMan separate init and new EXP apis
+    // separately. This must be done to avoid attempting to use symbols that do
+    // not exist in older loader runtimes.
+#ifdef _WIN32
```

Comment:
if these libraries are already part of the process, it should be possible to skip loading the library at all, and simply use:
```
dlsym(RTLD_DEFAULT, "zesSymbolName");
```
on Linux, and:
```
HMODULE hModule = GetModuleHandle(NULL);
FARPROC pFunc = GetProcAddress(hModule, "zesSymbolName");
````
on Windows.

But I won't insist, either solution works.

Reply:
To prevent multiple loading that would not be a bad idea, let me see if I can add that style to the common functions.

Reply:
I have changed this to use the current process handle vs additionally loading the loader.